clock/clock.go

```diff
@@ -30,13 +30,36 @@ type PassiveClock interface {
 // needs to do arbitrary things based on time.
 type Clock interface {
 	PassiveClock
+	// After returns the channel of a new Timer.
+	// This method does not allow to free/GC the backing timer before it fires. Use
+	// NewTimer instead.
 	After(d time.Duration) <-chan time.Time
+	// NewTimer returns a new Timer.
 	NewTimer(d time.Duration) Timer
+	// Sleep sleeps for the provided duration d.
+	// Consider making the sleep interruptible by using 'select' on a context channel and a timer channel.
 	Sleep(d time.Duration)
+	// Tick returns the channel of a new Ticker.
+	// This method does not allow to free/GC the backing ticker. Use
+	// NewTicker instead.
 	Tick(d time.Duration) <-chan time.Time
 }
 
-var _ = Clock(RealClock{})
+// V2 allows for injecting fake or real clocks into code that
+// needs to do arbitrary things based on time.
+type V2 interface {
```

Comment:
V2 is not a great name :-)

Reply:
Since the interfaces are pretty much frozen once built, I think you can call it "ClockWithTicker" or something like this.

Reply:
I've tried both `ClockV2` and `ClockWithTicker` but linted doesn't like such names - see https://github.com/kubernetes/utils/pull/197/checks?check_run_id=1685665492#step:4:62 That's why I tried `V2`, which you don't like :)

Reply:
Yeah, "WithTicker" is fine (to me at least ;-))

Reply:
@apelisse PTAL